File: MVC/MVC.go
package main

import "fmt"

type Student struct {
	name string
	rollNo int
}

func(s *Student)getName()string{
	return s.name
}

func(s *Student)setName(name string){
	s.name=name
}

func(s *Student)getRollNo()int{
	return s.rollNo
}

func(s *Student)setRollNo(rollNo int){
	s.rollNo=rollNo
}


type View struct {

}

func(v *View)printStudentDetails(name string,rollNo int){
	fmt.Println("Student:[name:",name,",rollNo:",rollNo,"]")
}


type Controller struct {
	model *Student
	view *View
}


func(s *Controller)getStudentName()string{
	return s.model.getName()
}

func(s *Controller)setStudentName(name string){
	s.model.setName(name)
}

func(s *Controller)getStudentRollNo()int{
	return s.model.getRollNo()
}

func(s *Controller)setStudentRollNo(rollNo int){
	s.model.setRollNo(rollNo)
}

func(s *Controller)updateView(){
	s.view.printStudentDetails(s.model.getName(),s.model.getRollNo())
}

func main(){
	studentModel:=new(Student)
	studentModel.setRollNo(10)
	studentModel.setName("Lucy")

	studentView:=new(View)

	studentController:=&Controller{studentModel,studentView}
	studentController.updateView()
	studentController.setStudentName("Joh")
	studentController.updateView()
}

File: MVC/MVC.js
/**
 * Created by jamesyan on 2017/3/25.
 */

var Student=function(name,rollNo){
    this.name=name;
    this.rollNo=rollNo;
}

Student.prototype.getName=function(){
    return this.name;
}

Student.prototype.setName=function(name){
    this.name=name;
}

Student.prototype.getRollNo=function(){
    return this.rollNo;
}

Student.prototype.setRollNo=function(rollNo){
    this.rollNo=rollNo;
}

var View=function(){
    this.printStudentDetails=function(name,rollNo){
        console.log("Student :[name:"+name+",rollNo:"+rollNo+"]");
    }
}


var Controller=function(model,view){
    this.model=model;
    this.view=view;
}

Controller.prototype.getStudentName=function(){
    return this.model.getName();
}

Controller.prototype.setStudentName=function(name){
    this.model.setName(name);
}

Controller.prototype.getStudentRollNo=function(){
    return this.model.getRollNo();
}

Controller.prototype.setStudentRollNo=function(rollNo){
    this.model.setRollNo(rollNo);
}

Controller.prototype.updateView=function(){
    this.view.printStudentDetails(this.model.getName(),this.model.getRollNo())
}








